sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -79,12 +79,13 @@ class CacheManager extends Logging {
       logWarning("Asked to cache already cached data.")
     } else {
       val sparkSession = query.sparkSession
+      val qe = sparkSession.sessionState.executePlan(planToCache)
       val inMemoryRelation = InMemoryRelation(
         sparkSession.sessionState.conf.useCompression,
         sparkSession.sessionState.conf.columnBatchSize, storageLevel,
-        sparkSession.sessionState.executePlan(planToCache).executedPlan,
+        qe.executedPlan,
         tableName,
-        planToCache)
+        qe.optimizedPlan)
```

Comment:
Hi, @jzhuge . Could you add a test case for this, too?

Reply:
Thanks @dongjoon-hyun for the review. Unfortunately I couldn't find a good way to unit test the accuracy of `InMemoryRelation.statsOfPlanToCache` and there is no existing unit test for this field. Any suggestion is welcome.

Reply:
I'm not sure that we can test this without filter pushdown in stats calculation.

We've added filter pushdown, so the stats methods use `PhysicalOperation` to detect scans paired with projections and filters. When getting the stats for a scan, those filters are passed so that we can get accurate stats. Without that optimization, the stats would be identical between the analyzed plan and the optimized plan.